### ISSUE TITLE
Reference path to .sst/types is not being added to sst-env.d.ts for StaticSite construct

### ISSUE THREAD

**Issue by DanielTatarkin** [sst/sst#3315]
## Overview
The following StaticSite construct line **only** adds `/// <reference types="vite/client"` to `sst-env.d.ts` by generating the file on each `sst deploy` run
https://github.com/sst/sst/blob/7e6e68d03696b27a5d5a566e4168a7b2bdb52178/packages/sst/src/constructs/StaticSite.ts#L494

This leads to an improper type reference detailed in the error below

## Error example
```shell 
SST v2.25.2

➜  App:     sst-app
   Stage:   prod
   Region:  us-east-1
   Account: XXXXXXX

Building static site packages/frontend

> @sst-app/frontend@0.0.0 build /Users/daniel/Developer/sst-app/packages/frontend
> vue-tsc && vite build

../core/src/dynamo.ts:10:16 - error TS2339: Property 'db' does not exist on type 'TableResources'.

10   table: Table.db.tableName,
```

## Workaround
As a temporary workaround, I had to add `/// <reference path="../../../.sst/types/index.ts" />` to `vite-env.d.ts` file which resolves the error and this file's content is not overwritten by `sst deploy`

## Proposal
I believe the true intention is to add `/// <reference path="${relPathToSstTypesFile}" />` to generated `sst-env.d.ts` as well

Similar to SsrSite https://github.com/sst/sst/blob/7e6e68d03696b27a5d5a566e4168a7b2bdb52178/packages/sst/src/constructs/SsrSite.ts#L1339
```typescript
private writeTypesFile() {
    const typesPath = path.resolve(
      this.props.path,
      this.buildConfig.typesPath,
      "sst-env.d.ts"
    );

    // Do not override the types file if it already exists
    if (fs.existsSync(typesPath)) return;

    const relPathToSstTypesFile = path.join(
      path.relative(path.dirname(typesPath), useProject().paths.root),
      ".sst/types/index.ts"
    );
    fs.writeFileSync(
      typesPath,
      `/// <reference path="${relPathToSstTypesFile}" />`
    );
  }
```

### current sst-env.d.ts
```typescript
/// <reference types="vite/client" />
interface ImportMetaEnv {
  readonly VITE_API_URL: string
}
interface ImportMeta {
  readonly env: ImportMetaEnv
}
```
### what I think sst-env.d.ts should be
```typescript
/// <reference path="../../../.sst/types/index.ts" />
/// <reference types="vite/client" />
interface ImportMetaEnv {
  readonly VITE_API_URL: string
}
interface ImportMeta {
  readonly env: ImportMetaEnv
}
```

**Comment by DanielTatarkin** [comment]
Good call out by Jay:

> Why is your static site accessing Table.db.tableName?

In my case I'm not accessing that prop directly but I was importing everything from `packages/core/*` in my Frontend's tsconfig in order to import other unrelated stuff. Perhaps I could look into refactoring/decoupling my logic and packages here.

**Comment by imjulianeral** [comment]
@DanielTatarkin I love your workaround, it helped me to fix my bundling error. My use case is similar to yours. I created an API with Hono following [this example](https://hono.dev/guides/rpc#using-with-larger-applications), since I added the dynamodb table and imported the `AppType` to my vite app, I got the exact same error as you.

Have you tried something else? I read on discord that you were going to decouple your types to fix it, but I don't know how to do it because the type is tightly coupled to my API, like your types are tightly coupled to your electrodb entity.
